src/adapter/graphql/src/mutations/flows_mut/dataset_flow_runs_mut.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use chrono::Utc;
+use kamu_core::CurrentAccountSubject;
+use kamu_flow_system::{FlowKeyDataset, FlowService, RequestFlowError};
+use opendatafabric as odf;
+
+use super::{
+    ensure_expected_dataset_kind,
+    ensure_scheduling_permission,
+    FlowIncompatibleDatasetKind,
+};
+use crate::prelude::*;
+use crate::LoggedInGuard;
+
+///////////////////////////////////////////////////////////////////////////////
+
+pub struct DatasetFlowRunsMut {
+    dataset_handle: odf::DatasetHandle,
+}
+
+#[Object]
+impl DatasetFlowRunsMut {
+    #[graphql(skip)]
+    pub fn new(dataset_handle: odf::DatasetHandle) -> Self {
+        Self { dataset_handle }
+    }
+
+    #[graphql(guard = "LoggedInGuard::new()")]
+    async fn trigger_flow(
+        &self,
+        ctx: &Context<'_>,
+        dataset_flow_type: DatasetFlowType,
+    ) -> Result<TriggerFlowResult> {
+        if let Some(e) =
+            ensure_expected_dataset_kind(ctx, &self.dataset_handle, dataset_flow_type).await?
+        {
+            return Ok(TriggerFlowResult::IncompatibleDatasetKind(e));
+        }
+
+        ensure_scheduling_permission(ctx, &self.dataset_handle).await?;
```

Comment:
Same here for permissions - shouldn't this be in the service?

Reply:
agreed on slack to postpone this and perhaps later extract this into a `DatasetFlowService`